File: problem5/Problem5.java
public class Problem5 {
	public static void main(String[] args) {
		System.out.println(smallest_divisible(20));
	}

	public static int smallest_divisible(int by_all_up_to) {
		int val = 0;

		while(true) {
			++val;
			if(divisible_by_all_up_to(val, by_all_up_to)) {
				return val;
			}
		}
	}

	public static boolean divisible_by_all_up_to(int val, int by_all_up_to) {
		for(int i = 2; i < by_all_up_to; ++i) {
			if(val % i != 0) {
				return false;
			}
		}

		return true;
	}
}

File: problem005/c/problem5.c
#include <stdio.h>
#include <math.h>
#include <prime.h>


int step_size(int by_all_upto) {
        int step = 1;
        for (int i = 2; i <= by_all_upto; ++i) {
                if (is_prime(i)) {
                        step *= i;
                }
        }
        return step;
}


int divisible_by_all_upto(int val, int upto) {
        int all = 1;
        for (int i = 2; i <= upto; ++i) {
                if(val % i != 0) {
                        return 0;
                }
        }
        return 1;
}


int smallest_divisible(int by_all_upto) {
        int step = step_size(by_all_upto);
        for(int n = step; ; n+=step) {
                if (divisible_by_all_upto(n, by_all_upto)) {
                        return n;
                }
        }
}


int main() {
        printf("%d\n", smallest_divisible(20));
	return 0;
}
